solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -991,6 +992,7 @@ public static class Builder {
     private ExecutorService executor;
     protected RequestWriter requestWriter;
     protected ResponseParser responseParser;
+    private volatile Set<String> queryParams = Collections.emptySet();
```

Comment:
I don't think this needs to be 'volatile' anymore

Reply:
definitely not volatile in a Builder!
Eric, I think you were merely copying the field definition in the client.  It's only volatile there because of the setters we are removing, in case two threads work with the client at the same time.

Reply:
we should make this set.of everywehre ;-)  And prevent the Collections.emptySet via forbidden apis...